examples/apps/staging/src/start.ts

```diff
@@ -43,31 +41,67 @@ const render = (groceryList: ISuggestionGroceryList) => {
 	debugRoot.render(createElement(DebugView, { groceryList }));
 };
 
-const tokenProvider = createInsecureTinyliciousTestTokenProvider();
-const urlResolver = createInsecureTinyliciousTestUrlResolver();
-const codeLoader = new StaticCodeLoader(new GroceryListContainerRuntimeFactory());
-
 async function start(): Promise<void> {
+	const service = getSpecifiedServiceFromWebpack();
+	const {
+		urlResolver,
+		documentServiceFactory,
+		createCreateNewRequest,
+		createLoadExistingRequest,
+	} = await createExampleDriver(service);
+
+	const codeLoader = new StaticCodeLoader(new GroceryListContainerRuntimeFactory());
+
 	let id: string;
 	let groceryList: ISuggestionGroceryList;
 
 	if (location.hash.length === 0) {
+		// Some services support or require specifying the container id at attach time (local, odsp). For
+		// services that do not (t9s), the passed id will be ignored.
+		id = Date.now().toString();
+		const createNewRequest = createCreateNewRequest(id);
 		const container = await createDetachedContainer({
 			codeDetails: { package: "1.0" },
 			urlResolver,
-			documentServiceFactory: createRouterliciousDocumentServiceFactory(tokenProvider),
+			documentServiceFactory,
 			codeLoader,
 		});
 		groceryList = (await container.getEntryPoint()) as ISuggestionGroceryList;
-		await container.attach(createTinyliciousTestCreateNewRequest());
-		if (container.resolvedUrl === undefined) {
-			throw new Error("Resolved Url not available on attached container");
+		await container.attach(createNewRequest);
+		// For most services, the id on the resolvedUrl is the authoritative source for the container id
+		// (regardless of whether the id passed in createCreateNewRequest is respected or not). However,
+		// for odsp the id is a hashed combination of drive and container ID which we can't use. Instead,
+		// we retain the id we generated above.
+		if (service !== "odsp") {
+			if (container.resolvedUrl === undefined) {
+				throw new Error("Resolved Url not available on attached container");
+			}
+			id = container.resolvedUrl.id;
 		}
-		id = container.resolvedUrl.id;
 	} else {
 		id = location.hash.substring(1);
 		const container = await loadExistingContainer({
-			request: { url: id },
+			request: await createLoadExistingRequest(id),
+			urlResolver,
+			documentServiceFactory,
+			codeLoader,
+		});
+		groceryList = (await container.getEntryPoint()) as ISuggestionGroceryList;
+	}
+
+	render(groceryList);
+	updateTabForId(id);
+}
+
+try {
+	await start();
+} catch (error) {
+	console.error(error);
+	console.log(
+		"%cEnsure you are running the appropriate Fluid Server for the selected service\nUse:`npm run start:server` for local, or configure ODSP for odsp",
+		"font-size:30px",
+	);
+}
```

Comment:
Remove this try/catch block, the call to start() below is sufficient.

Also there are some duplicated lines below that need to be cleaned up.